src/cache.ts

```diff
@@ -8,17 +8,25 @@ interface DecodedToken {
   user: any;
 }
 
-interface CacheEntry extends CacheKeyData {
+interface CacheEntry {
   id_token: string;
   access_token: string;
   expires_in: number;
   decodedToken: DecodedToken;
+  audience: string;
+  scope: string;
 }
 
 interface CachedTokens {
   [key: string]: CacheEntry;
 }
 
+export interface ICache {
+  save(entry: CacheEntry): void;
+  get(key: CacheKeyData): CacheEntry;
+  clear(): void;
+}
+
 const createKey = (e: CacheKeyData) => `${e.audience}::${e.scope}`;
```

Comment:
Are we sure this is unique in both memory and localstorage given how they are partitioned that multiple tenants and apps won't collide?

Reply:
I had added a key prefix of `@@auth0spajs@@` to the local storage implementation as I needed a way to be able to pick out the Auth0 cache keys during `clear`. I've extended this to the memory implementation now as well.

Reply:
Consider including clientId to prevent collisions.

Reply:
This is now done.